data/schema/mapper.go

```diff
@@ -13,11 +15,16 @@ const ErrEventTime = "unable to parse event time"
 
 func (s *BasalSample) MapForAutomatedBasal(event *automated.Automated) error {
 	var err error
+	// assign an uuid for keeping a link between the two collection
+	event.InternalID = uuid.New().String()
```

Comment:
I am not sure to understand why we need to set a UUID to `event` and not directly to `s`? :thinking:

Reply:
for writing the UUID in both data that will be inserted in the two collection